ts/types/CreditCard.ts

```diff
@@ -29,14 +29,20 @@ export type CreditCardType =
   | "POSTEPAY"
   | "UNKNOWN";
 
-const CreditCardUnknown = t.type(
-  {
-    id: CreditCardId,
-    lastUsage: NonEmptyString,
-    pan: PatternString("^[0-9]+$"),
-    owner: NonEmptyString,
-    expirationDate: NonEmptyString
-  },
+const CreditCardUnknownR = t.type({
```

Comment:
these temporary types should go away, we should use the types generated from the swagger definitions:

![image](https://user-images.githubusercontent.com/48206/42109769-19e938e8-7bdf-11e8-8b72-de321fe33e15.png)

Reply:
The problem with this type (and all the other generated type definitions) is that all fields are marked as optional -- which I believe shouldn't be the case. Would it make sense to have a discussion about which fields should be mandatory and which shouldn't, and then update the definitions to reflect this?

Reply:
you can let the generator generate all the types from the specs, having all fields optional (say type `CreditCard`) - then you can derive you own "all required" type:

```ts
type AllRequiredCreditCard = Required<CreditCard>;
```

We can have that discussion later, in the 2nd milestone of the project.